### PR TITLE
txn: fix UpdateForUpdateTS for non autocommit pessimistic mode

### PR DESCRIPTION
**Pull request by cfzjywxk**
### What problem does this PR solve? 

`SelectLock` executor will still need to call `UpdateForUpdateTS` if its sub plan is not point get.  But for pessimistic transactions with `autocommit == 0`, the transaction may not be activated yet. And the execution will get `[kv:8024]invalid transaction` error,  see the test case for example.

### What is changed and how it works?

Seems it's needed to make sure the transaction is activated if `UpdateForUpdateTS`  is called

### Check List 

Tests 

 - Unit test

Code changes

Side effects

Related changes

Release note

### CODE REVIEW

File: executor/builder.go [hunk 1]
```diff
@@ -1633,6 +1633,18 @@ func (b *executorBuilder) updateForUpdateTSIfNeeded(selectPlan plannercore.Physi
...
+	txn, err := b.ctx.Txn(false)
+	if err != nil {
+		return err
+	}
+	if !txn.Valid() {
+		txn, err := b.ctx.Txn(true)
+		if err != nil {
+			return err
+		}
+		return UpdateForUpdateTS(b.ctx, txn.StartTS())
```
Comment (coocood) [inline executor/builder.go]: We don't need to call it, the `ForUpdateTS` is equal to startTs.

**Comment by coocood** [comment]
LGTM

**Comment by jackysp** [comment]
PTAL @lysu

**Comment by lysu** [comment]
@cfzjywxk do we need cherry-pick this bug?

**Comment by cfzjywxk** [comment]
> @cfzjywxk do we need cherry-pick this bug?

3.0 version `UpdateForUpdateTS` will activate txn, it's ok on 3.0

**Pull request merged**